src/python/py/models/postprocess_lm_head_tied_embeddings.py

```diff
@@ -0,0 +1,313 @@
+        final_reshape_node
+    ]
+    
+    # Modify this part to handle asymmetric quantization if needed
+    if use_zero_points:
+        # Gather the zero points
+        gathered_zero_points = "gathered_zero_points"
+        gather_zero_points_node = helper.make_node(
+            'Gather',
+            inputs=[zero_points_name, input_ids],
+            outputs=[gathered_zero_points],
+            name='/model/embed_tokens/GatherZeroPoints',
+            axis=0
+        )
+        
+        # Unsqueeze zero points for broadcasting
+        unsqueeze_zp_node = helper.make_node(
+            'Unsqueeze',
+            inputs=[gathered_zero_points, "scales_axes"],
+            outputs=["unsqueezed_zero_points"],
+            name='/model/embed_tokens/UnsqueezeZeroPoints'
+        )
+        
+        # Cast zero points to float
+        cast_gathered_zp_node = helper.make_node(
+            'Cast',
+            inputs=["unsqueezed_zero_points"],
+            outputs=["casted_gathered_zero_point"],
+            name='/model/embed_tokens/CastGatheredZeroPoint',
+            to=scale_value_type
+        )
+        
+        # Replace the standard zero_point subtraction with the gathered one
+        sub_node.input[1] = "casted_gathered_zero_point"
+        
+        # Insert the new nodes
+        new_nodes.insert(2, gather_zero_points_node)
+        new_nodes.insert(3, unsqueeze_zp_node)
+        new_nodes.insert(6, cast_gathered_zp_node)
+    
+    graph.node.extend(new_nodes)
+    
+    print("Successfully tied embedding weights to quantized LM head weights using Cast+Mul operations")
+
+
+def get_node_attribute(node: onnx.NodeProto, attribute_name: str):
+    for attr in node.attribute:
+        if attr.name == attribute_name:
+            value = onnx.helper.get_attribute_value(attr)
+            return value
+    return None
+
+
+def find_graph_input(graph, input_name):
+    for input in graph.input:
+        if input.name == input_name:
+            return input
+    return None
+
+
+def find_graph_output(graph, output_name):
+    for output in graph.output:
+        if output.name == output_name:
+            return output
+    return None
+
+
+def get_tensor_type_from_graph(graph, tensor_name: str):
+    tensor_type_map = {obj.name: obj.type for obj in graph.value_info}
+
+    if tensor_name in tensor_type_map:
+        return tensor_type_map[tensor_name].tensor_type
+    
+    g_input = find_graph_input(graph, tensor_name)
+    if g_input:
+        return g_input.type.tensor_type
+
+    g_output = find_graph_output(graph, tensor_name)
+    if g_output:
+        return g_output.type.tensor_type
+
+    return None
+
+
+def convert_gather_to_use_lm_head_weights(model_path, output_path, load_external_data=True):
+    # Load the ONNX model
+    print(f"Loading model from {model_path}...")
+    model_name = "model.onnx"
+    model = onnx.load(model_path + model_name, load_external_data=False)
+    if load_external_data:
```

Comment:
Shall we just use `load_external_data=True` in onnx.load? 
Why need we have an option not loading external data.

Reply:
Sure